mailbox/api/src/main/java/org/apache/james/mailbox/MailboxManager.java

```diff
@@ -333,7 +333,7 @@ public final int hashCode() {
      *          the context for this call, not null
      * @return  a list of MessageId of messages belong to that Thread
      */
-    Flux<MessageId> getThread(ThreadId threadId, MailboxSession session) throws MailboxException;
+    Publisher<MessageId> getThread(ThreadId threadId, MailboxSession session) throws MailboxException;
```

Comment:
Methods with publishers should not throw.

They should return publisher that eventually completes with an error (`Flux.error`)

The error is returned when blocking the publisher not while generating the publisher.

Reply:
This method depends on MailboxManager::search which throws MailboxException. I guess there is nothing I can do with that.

Reply:
I think those search methods were not reactive before and that throws stayed in the api interface. You can remove it I think. If you look at the current implementations of MailboxManager, they don't seem to throw it either

Reply:
And yes there is other publisher methods still doing this too... we should clean up a bit :)